eng/pipelines/test-windows-job-single-machine.yml

```diff
@@ -35,6 +35,14 @@ jobs:
   steps:
     - checkout: none
 
+    # A .NET 6 runtime is needed since our unit tests target net6.0
```

Comment:
Should we just be upgrading to something newer here?

Reply:
We could update our tests to target net7.0, but that seems like a more disruptive change. In the scheme of things the SingleMachine CI jobs aren't strictly necessary and may not be providing value over the Helix CI. They were mainly intended to ensure our infra changes didn't break the ability to run tests locally.

Reply:
FWIW i was chatting with @dibarbet about this the other day. 

> They were mainly intended to ensure our infra changes didn't break the ability to run tests locally.

Exactly. More broadly it was to make sure that we had some sanity check in place to make sure people could develop locally: build, test and run. We lost that to a large extent when we moved to Helix. The single machine tests exists to make sure we can develop locally.

That means we don't even need to run all tests here. We just need to run enough that we're convinced we haven't broken things.